src/DictList.elm

```diff
@@ -938,6 +965,174 @@ fromDict dict =
+-}
+inits : DictList comparable v -> List (DictList comparable v)
+-- @FIXME
+inits list =
+  toList list
+    |> List.Extra.inits
+    |> List.map fromList
+
+{-| Returns a list of repeated applications of `f`.
+
+If `f` returns `Nothing` the iteration will stop. If it returns `Just y` then `y` will be added to the list and the iteration will continue with `f y`.
+    nextYear : Int -> Maybe Int
+    nextYear year =
+      if year >= 2030 then
+        Nothing
+      else
+        Just (year + 1)
+    -- Will evaluate to [2010, 2011, ..., 2030]
+    iterate nextYear 2010
+-}
+iterate : ((comparable, v) -> Maybe (comparable, v)) -> (comparable, v) -> DictList comparable v
+iterate f x =
+  List.Extra.iterate f x
+  |> fromList
+
+{-| Decompose a list into its head and tail. If the list is empty, return `Nothing`. Otherwise, return `Just (x, xs)`, where `x` is head and `xs` is tail.
+
+    uncons (fromList [(1, 1),(2, 2),(3, 3)] == Just ((1, 1), fromList [(2, 2), (3, 3)])
+    uncons empty = Nothing
+-}
+uncons : DictList comparable v -> Maybe ( (comparable, v), DictList comparable v)
+uncons xs =
+  toList xs
+    |> List.Extra.uncons
+    |> Maybe.map (\(a, la) -> (a, fromList la))
+
+{-| Find the first maximum element in a list using a comparable transformation
+-}
+maximumBy : (comparable2 -> a -> comparable1) -> DictList comparable2 a -> Maybe (comparable2, a)
+maximumBy f ls =
+  toList ls
+    |> List.Extra.maximumBy (uncurry f)
+
+{-| Find the first minimum element in a list using a comparable transformation
+-}
+minimumBy : (comparable2 -> a -> comparable1) -> DictList comparable2 a -> Maybe (comparable2, a)
+minimumBy f ls =
+  toList ls
+    |> List.Extra.minimumBy (uncurry f)
+
+{-| Take elements in order as long as the predicate evaluates to `True`
+-}
+takeWhile : ((comparable, a) -> Bool) -> DictList comparable a -> DictList comparable a
+takeWhile predicate xs =
+  toList xs
+    |> List.Extra.takeWhile predicate
+    |> fromList
+
+{-| Drop elements in order as long as the predicate evaluates to `True`
+-}
+dropWhile : ((comparable, a) -> Bool) -> DictList comparable a -> DictList comparable a
+dropWhile predicate list =
+  toList list
+    |> List.Extra.dropWhile predicate
+    |> fromList
+
+{-| Remove duplicate values, keeping the first instance of each element which appears more than once.
+
+    unique [0,1,1,0,1] == [0,1]
+-}
+unique : DictList comparable1 comparable2 -> DictList comparable1 comparable2
+unique list =
+  toList list
+    |> List.Extra.unique
+    |> fromList
+
+{-| Drop duplicates where what is considered to be a duplicate is the result of first applying the supplied function to the elements of the list.
+-}
+uniqueBy : (comparable1 -> a -> comparable2) -> DictList comparable1 a -> DictList comparable1 a
+uniqueBy f list =
+  toList list
+    |> List.Extra.uniqueBy (uncurry f)
+    |> fromList
```

Comment:
This one, on the other hand, could remove some values, so it could do something. I wonder which value the implementation retains (since the "duplicate" values are not, themselves, necessarily equal)? That would be worth documenting (I suppose `List.Extra` might not document it).